### PR TITLE
chore: new package structure with src  dir, vitest, deps updates, revamped actions and more

### PR DESCRIPTION
**Pull request by dipankarmaikap**
This PR introduces a new structure and multiple improvements, including:

- Moved to `src` directory instead of `lib`.
- Refactored project under `pnpm` workspaces.
- Cypress config and folder moved to root.
- Default E2E runs against `playground/test` (more coverage needed).
- Added Vitest function tests (more tests needed).
- Updated GitHub Actions with `commitlint` and separate workflows for lint/test.
- Upgraded dependencies (no breaking changes to SDK).

### CODE REVIEW

**Review by manuelschroederdev — APPROVED** [review]

**Review by alvarosabu — CHANGES_REQUESTED** [review]
Doing the QG

**Review by alvarosabu — CHANGES_REQUESTED** [review]
Hi @dipankarmaikap amazing job you did here. Just one minor blocking point, I also added some commit suggestion to ease the changes required. Let me know if you have any questions or something you need support with

File: playground/ssg/astro.config.mjs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        'svelte_counter': 'storyblok/SvelteCounter',
+        'react_counter': 'storyblok/ReactCounter',
+        'new-component': 'storyblok/NewComponent',
+        'featured-articles': 'storyblok/FeaturedArticles',
+      },
+    }),
+    tailwind(),
+  ],
+  vite: {
+    plugins: [mkcert()],
```
[the remark this answers is not in the document]
Comment (dipankarmaikap) [reply playground/ssg/astro.config.mjs:35]: If we do this we get the following error
```js
Unknown file extension ".astro" for ..../dist/components/StoryblokComponent.astro
You likely need to add this package to `vite.ssr.noExternal` in your astro config file.
```

File: playground/ssg/astro.config.mjs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import tailwind from '@astrojs/tailwind';
+import mkcert from 'vite-plugin-mkcert';
+
+export default defineConfig({
+  integrations: [
+    svelte(),
+    vue(),
+    react(),
+    storyblok({
+      accessToken: 'OsvNv534kS2nivAAj1EPVgtt',
```
[the remark this answers is not in the document]
Comment (manuelschroederdev) [reply playground/ssg/astro.config.mjs:15]: Hey @alvarosabu, I'd suggest we keep it as it is for this PR and tackle replacing the tokens with env variables in a subsequent step for all frameworks? :-)

**Review by alvarosabu — APPROVED** [review]

**Pull request merged**